man/tpm2_hierarchycontrol.1.md

```diff
@@ -0,0 +1,77 @@
+% tpm2_hierarchycontrol(1) tpm2-tools | General Commands Manual
+%
+% July 2019
+
+# NAME
+
+**tpm2_hierarchycontrol**(1) - Enable and disable use of a hierarchy and its
+associated NV storage.
+
+# SYNOPSIS
+
+**tpm2_hierarchycontrol** [*OPTIONS*] _OPERATION_
+
+# DESCRIPTION
+
+**tpm2_hierarchycontrol**(1) - Allows user change phEnable, phEnableNV,
+shEnable and ehEnable when the proper authorization is provided. Authorization
+ should be one out of owner hierarchy auth, endorsement hierarchy auth and
+ platform hierarchy auth. **As an argument the tool takes the \_OPERATION\_ as
+ string clear|set to clear or set the TPMA_STARTUP_CLEAR bits.**
+Note: If password option is missing, assume NULL.
+
+# OPTIONS
+
+  * **-c**, **\--context-object**=_CONTEXT_:
+
+    Context object for the loaded object. Either a file or a handle number.
+    See section "Context Object Format".
+
+  * **-p**, **\--auth-key**=_KEY\_AUTH_:
+
+    Optional authorization value to use the key specified by **-c**.
+    Authorization values should follow the "authorization formatting standards",
+    see section "Authorization Formatting".
+
+  * **-e**, **\--enable**=_TPMA\_STARTUP\_CLEAR\_BITS:
```

Comment:
kill this option and make it argument.
So
```
tpm2_hierarchycontrol -c e -e phEnable set
```
becomes:
```sh
tpm2_hierarchycontrol -c e phEnable set
```

Reply:
I would also accept:
```
phEnable=set|1
phEnable=clear|0
```
Sample Code:
```C
#include <string.h>

int main(int argc, char *argv[]) {

  char *split = strchr(argv[1], '=');
  if (!split) {
    return 1;
  }

  char *key=argv[1];
  *split = '\0';
  char *value = split + 1;

  printf("%s=%s\n", key, value);

  return 0;
}
```

Reply:
I will use first one.